ecosystem/typescript/sdk_v2/src/core/account.ts

```diff
@@ -49,17 +56,44 @@ export class Account {
 
     // Derive the public key from the private key
     this.publicKey = privateKey.publicKey();
+
+    // Derive the signing scheme from the public key
+    if (this.publicKey instanceof Ed25519PublicKey) {
+      this.signingScheme = SigningScheme.Ed25519;
+    } else if (this.publicKey instanceof MultiEd25519PublicKey) {
+      this.signingScheme = SigningScheme.MultiEd25519;
+    } else if (this.publicKey instanceof Secp256k1PublicKey) {
+      // Secp256k1
+      this.signingScheme = SigningScheme.Secp256k1Ecdsa;
+    } else {
+      throw new Error("Can not create new Account, unsupported public key type");
+    }
+
     this.privateKey = privateKey;
     this.accountAddress = address;
   }
 
   /**
    * Derives an account with random private key and address
    *
+   * @param args.scheme SigningScheme - type of SigningScheme to use.
+   * Currently only Ed25519, MultiEd25519, and Secp256k1 are supported
+   *
    * @returns Account
    */
-  static generate(): Account {
-    const privateKey = Ed25519PrivateKey.generate();
+  static generate(args: { scheme: SigningScheme }): Account {
```

Comment:
thinking whether the scheme args should be optional, and we set one by default. @davidbarsky @gregnazario @xbtmatt for thoughts also